src/dictionary.h

```diff
@@ -27,22 +38,86 @@ typedef struct dictionary {
 
     uint8_t flags; ///< DICTIONARY_FLAG_*
 
-    struct dictionary_stats *stats; ///< statistics of this dictionary
-    pthread_rwlock_t *rwlock;       ///< lock for synchronizing access to this
+    struct dictionary_stats *stats; ///< Statistics of this dictionary. This may be NULL.
+    pthread_rwlock_t *rwlock;       ///< Lock for synchronizing access to this. This may be NULL.
 } DICTIONARY;
 
-#define DICTIONARY_FLAG_DEFAULT                 0x00000000
-#define DICTIONARY_FLAG_SINGLE_THREADED         0x00000001
-#define DICTIONARY_FLAG_VALUE_LINK_DONT_CLONE   0x00000002
-#define DICTIONARY_FLAG_NAME_LINK_DONT_CLONE    0x00000004
-#define DICTIONARY_FLAG_WITH_STATISTICS         0x00000008
+#define DICTIONARY_FLAG_DEFAULT                 0x00000000 ///< no specific meaning.
+#define DICTIONARY_FLAG_SINGLE_THREADED         0x00000001 ///< If enabled synchronize access to dictionary.
```

Comment:
if set, do not use the avl *_lock() functions, but the plain ones. This can lead to crashes if multiple threads are adding, removing and reading the tree concurrently.